DNN gets stuck when database is unavailable during initialization

## Description of bug
When the application starts up, if the database connection fails, the application will continue showing the error <q>Connection To The Database Failed</q> even if the database becomes available.

## Steps to reproduce
1. Install DNN
2. Turn off the database
3. Navigate to DNN (see error message)
4. Turn on the database
5. Click <q>Return to Site</q> button/link

## Current behavior
Continue seeing error message

## Expected behavior
Application starts up correctly

## Screenshots
![Screenshot: DNN Error: Connection To The Database Failed](https://user-images.githubusercontent.com/59507/88305951-bda19000-cccf-11ea-885f-bb9995adeef6.png)

## Error information
<details>
<summary>Portals/_default/Logs/2020.07.23.log.resources</summary>

> 2020-07-23 10:28:30.723-05:00 [DESKTOP-ULJFF6L][D:2][T:7][ERROR] DotNetNuke.Common.Initialize - The connection to the database has failed, however, the application is already completely installed, a 500 error page will be shown to visitors
> 2020-07-23 10:30:07.887-05:00 [DESKTOP-ULJFF6L][D:2][T:10][ERROR] DotNetNuke.Data.PetaPoco.PetaPocoHelper - [3] Error executing SQL: ;Exec dbo.GetEventLogType
> A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - The wait operation timed out.)
> 2020-07-23 10:30:07.891-05:00 [DESKTOP-ULJFF6L][D:2][T:10][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.Data.SqlClient.SqlException (0x80131904): A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - The wait operation timed out.) ---> System.ComponentModel.Win32Exception (0x80004005): The wait operation timed out
>    at System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString connectionOptions, SqlCredential credential, Object providerInfo, String newPassword, SecureString newSecurePassword, Boolean redirectedUserInstance, SqlConnectionString userConnectionOptions, SessionData reconnectSessionData, DbConnectionPool pool, String accessToken, Boolean applyTransientFaultHandling, SqlAuthenticationProviderManager sqlAuthProviderManager)
>    at System.Data.SqlClient.SqlConnectionFactory.CreateConnection(DbConnectionOptions options, DbConnectionPoolKey poolKey, Object poolGroupProviderInfo, DbConnectionPool pool, DbConnection owningConnection, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionFactory.CreatePooledConnection(DbConnectionPool pool, DbConnection owningObject, DbConnectionOptions options, DbConnectionPoolKey poolKey, DbConnectionOptions userOptions)
>    at System.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
>    at System.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject, DbConnectionOptions userOptions, DbConnectionInternal oldConnection)
>    at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, UInt32 waitForMultipleObjectsTimeout, Boolean allowCreate, Boolean onlyOneCheckConnection, DbConnectionOptions userOptions, DbConnectionInternal& connection)
>    at System.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject, TaskCompletionSource\`1 retry, DbConnectionOptions userOptions, DbConnectionInternal& connection)
>    at System.Data.ProviderBase.DbConnectionFactory.TryGetConnection(DbConnection owningConnection, TaskCompletionSource\`1 retry, DbConnectionOptions userOptions, DbConnectionInternal oldConnection, DbConnectionInternal& connection)
>    at System.Data.ProviderBase.DbConnectionInternal.TryOpenConnectionInternal(DbConnection outerConnection, DbConnectionFactory connectionFactory, TaskCompletionSource\`1 retry, DbConnectionOptions userOptions)
>    at System.Data.SqlClient.SqlConnection.TryOpenInner(TaskCompletionSource\`1 retry)
>    at System.Data.SqlClient.SqlConnection.TryOpen(TaskCompletionSource\`1 retry)
>    at System.Data.SqlClient.SqlConnection.Open()
>    at PetaPoco.Database.OpenSharedConnection()
>    at DotNetNuke.Data.PetaPoco.PetaPocoExt.ExecuteReader(Database database, String sql, Object[] args)
>    at DotNetNuke.Data.PetaPoco.PetaPocoHelper.ExecuteReader(String connectionString, CommandType type, Int32 timeoutSec, String sql, Object[] args)
>    at DotNetNuke.Data.SqlDatabaseConnectionProvider.ExecuteReader(String connectionString, CommandType commandType, Int32 commandTimeout, String procedureName, Object[] commandParameters)
>    at DotNetNuke.Data.SqlDataProvider.ExecuteReader(String procedureName, Object[] commandParameters)
>    at DotNetNuke.Data.DataProvider.GetLogTypeInfo()
>    at DotNetNuke.Services.Log.EventLog.DBLoggingProvider.<>c.<GetLogTypeInfo>b__19_0(CacheItemArgs c)
>    at DotNetNuke.Common.Utilities.DataCache.GetCachedDataFromRuntimeCache(CacheItemArgs cacheItemArgs, CacheItemExpiredCallback cacheItemExpired)
> ClientConnectionId:00000000-0000-0000-0000-000000000000
> Error Number:258,State:0,Class:20
> 2020-07-23 10:30:07.964-05:00 [DESKTOP-ULJFF6L][D:2][T:10][ERROR] DotNetNuke.Data.PetaPoco.PetaPocoHelper - [3] Error executing SQL: ;Exec dbo.GetEventLogType
> A network-related or instance-specific error occurred while establishing a connection to SQL Server. The server was not found or was not accessible. Verify that the instance name is correct and that SQL Server is configured to allow remote connections. (provider: TCP Provider, error: 0 - The wait operation timed out.)

</details>

## Affected version
<!-- 
Please add X in at least one of the boxes as appropriate. In order for an issue to be accepted, a developer needs to be able to reproduce the issue on a currently supported version. If you are looking for a workaround for an issue with an older version, please visit the forums at https://dnncommunity.org/forums
-->
* [ ] 10.00.00 alpha build
* [x] 09.06.02 latest supported release

--- a/DNN Platform/Library/Common/Globals.cs
+++ b/DNN Platform/Library/Common/Globals.cs
@@ -498,81 +498,81 @@ namespace DotNetNuke.Common
         {
             get
             {
-                if (_status == UpgradeStatus.Unknown)
+                if (_status != UpgradeStatus.Unknown && _status != UpgradeStatus.Error)
                 {
-                    var tempStatus = UpgradeStatus.Unknown;
+                    return _status;
+                }
 
-                    Logger.Trace("Getting application status");
-                    tempStatus = UpgradeStatus.None;
+                Logger.Trace("Getting application status");
+                var tempStatus = UpgradeStatus.None;
 
-                    // first call GetProviderPath - this insures that the Database is Initialised correctly
-                    // and also generates the appropriate error message if it cannot be initialised correctly
-                    string strMessage = DataProvider.Instance().GetProviderPath();
+                // first call GetProviderPath - this insures that the Database is Initialised correctly
+                // and also generates the appropriate error message if it cannot be initialised correctly
+                string strMessage = DataProvider.Instance().GetProviderPath();
 
-                    // get current database version from DB
-                    if (!strMessage.StartsWith("ERROR:"))
+                // get current database version from DB
+                if (!strMessage.StartsWith("ERROR:"))
+                {
+                    try
                     {
-                        try
-                        {
-                            _dataBaseVersion = DataProvider.Instance().GetVersion();
-                        }
-                        catch (Exception ex)
-                        {
-                            Logger.Error(ex);
-                            strMessage = "ERROR:" + ex.Message;
-                        }
+                        _dataBaseVersion = DataProvider.Instance().GetVersion();
                     }
-
-                    if (strMessage.StartsWith("ERROR"))
+                    catch (Exception ex)
                     {
-                        if (IsInstalled())
-                        {
-                            // Errors connecting to the database after an initial installation should be treated as errors.
-                            tempStatus = UpgradeStatus.Error;
-                        }
-                        else
-                        {
-                            // An error that occurs before the database has been installed should be treated as a new install
-                            tempStatus = UpgradeStatus.Install;
-                        }
+                        Logger.Error(ex);
+                        strMessage = "ERROR:" + ex.Message;
                     }
-                    else if (DataBaseVersion == null)
+                }
+
+                if (strMessage.StartsWith("ERROR"))
+                {
+                    if (IsInstalled())
                     {
-                        // No Db Version so Install
-                        tempStatus = UpgradeStatus.Install;
+                        // Errors connecting to the database after an initial installation should be treated as errors.
+                        tempStatus = UpgradeStatus.Error;
                     }
                     else
                     {
-                        var version = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version;
-                        if (version.Major > DataBaseVersion.Major)
-                        {
-                            // Upgrade Required (Major Version Upgrade)
-                            tempStatus = UpgradeStatus.Upgrade;
-                        }
-                        else if (version.Major == DataBaseVersion.Major && version.Minor > DataBaseVersion.Minor)
-                        {
-                            // Upgrade Required (Minor Version Upgrade)
-                            tempStatus = UpgradeStatus.Upgrade;
-                        }
-                        else if (version.Major == DataBaseVersion.Major && version.Minor == DataBaseVersion.Minor &&
-                                 version.Build > DataBaseVersion.Build)
-                        {
-                            // Upgrade Required (Build Version Upgrade)
-                            tempStatus = UpgradeStatus.Upgrade;
-                        }
-                        else if (version.Major == DataBaseVersion.Major && version.Minor == DataBaseVersion.Minor &&
-                                 version.Build == DataBaseVersion.Build && IncrementalVersionExists(version))
-                        {
-                            // Upgrade Required (Build Version Upgrade)
-                            tempStatus = UpgradeStatus.Upgrade;
-                        }
+                        // An error that occurs before the database has been installed should be treated as a new install
+                        tempStatus = UpgradeStatus.Install;
                     }
-
-                    _status = tempStatus;
-
-                    Logger.Trace(string.Format("result of getting providerpath: {0}", strMessage));
-                    Logger.Trace("Application status is " + _status);
                 }
+                else if (DataBaseVersion == null)
+                {
+                    // No Db Version so Install
+                    tempStatus = UpgradeStatus.Install;
+                }
+                else
+                {
+                    var version = System.Reflection.Assembly.GetExecutingAssembly().GetName().Version;
+                    if (version.Major > DataBaseVersion.Major)
+                    {
+                        // Upgrade Required (Major Version Upgrade)
+                        tempStatus = UpgradeStatus.Upgrade;
+                    }
+                    else if (version.Major == DataBaseVersion.Major && version.Minor > DataBaseVersion.Minor)
+                    {
+                        // Upgrade Required (Minor Version Upgrade)
+                        tempStatus = UpgradeStatus.Upgrade;
+                    }
+                    else if (version.Major == DataBaseVersion.Major && version.Minor == DataBaseVersion.Minor &&
+                             version.Build > DataBaseVersion.Build)
+                    {
+                        // Upgrade Required (Build Version Upgrade)
+                        tempStatus = UpgradeStatus.Upgrade;
+                    }
+                    else if (version.Major == DataBaseVersion.Major && version.Minor == DataBaseVersion.Minor &&
+                             version.Build == DataBaseVersion.Build && IncrementalVersionExists(version))
+                    {
+                        // Upgrade Required (Build Version Upgrade)
+                        tempStatus = UpgradeStatus.Upgrade;
+                    }
+                }
+
+                _status = tempStatus;
+
+                Logger.Trace(string.Format("result of getting providerpath: {0}", strMessage));
+                Logger.Trace("Application status is " + _status);
 
                 return _status;
             }

